browser/src/canvas/CanvasSectionContainer.ts

```diff
@@ -1772,6 +1773,13 @@ class CanvasSectionContainer {
 	}
 
 	private drawSections (frameCount: number = null, elapsedTime: number = null, tileSubset: Set<any> = null) {
+
+//              Un-comment to debug duplicate rendering problems:
+//		const stack = new Error().stack;
+//		console.log("Draw sections:\n", stack);
```

Comment:
Just a note, `console.trace` is a shorter/cleaner way of doing this.

Reply:
Sure - however our commercial users get scared by un-conditional log output - something users can look at; so we don't do that by default - cf. the window.app.console.log wrappers (which are themselves surprisingly slow ;-).
I also assume that new Error() - is unfeasibly expensive as regards stack unwinding - so well worth not doing unconditionally =)

Reply:
I'm not suggesting uncommenting it :) Just that the two lines are broadly equivalent to just `console.trace('Draw sections');`.

Reply:
Oh - and you're 100% right on requestReDraw - this needs to be done in an animationFrame callback - but only if we have the tiles / content de-hydrated I think. The problem is synchronizing the DOM updates with that (I expect) which are somewhat scattered.